### ISSUE TITLE
Zapier OAuth2 test returns "Unable to fetch access token: {}"

### ISSUE THREAD

**Issue by se468** [zapier/zapier-platform-cli#308]
Here are the steps to reproduce:

1. Install a new app with Zapier CLI
2. Set environments `zapier env 1.0.0` and then change the urls in `authentication.js`.
3. Run `zapier test`

The authorization works in production when I do `zapier push`

I looked at the test file and it looks like some of the values are hard coded. How can I change the tests so that it will pass again?

**Comment by eliangcs** [comment]
@se468 to make the test pass, I think you need to add environment variables to your local `.env` file instead of running `zapier env 1.0.0`. The `zapier env` sets the environment variables on the production server while `zapier test` only uses the ones in `.env` file.

**Comment by se468** [comment]
Thanks I've changed it to correct env variables and now it's testing the correct urls and environments. 

It's giving "Client authentication failed" errors now, I think it's because of the hardcoded values within the tests file.  Do I need to change the hardcoded `one_time_code` and `a_token` and `refresh_token` as well?

**Comment by eliangcs** [comment]
@se468 yes, you need to change those hardcoded values. If you don't want to hardcode those, you can use things like `process.env.REFRESH_TOKEN` like how we do with `ACCESS_TOKEN`.
